include/resolver.hpp

```diff
@@ -23,19 +26,37 @@ namespace kyrylokupin::asio::dns {
 
         template<qtype T>
         auto query(const std::string domain) -> asio::awaitable<std::vector<dns_answer<T>>> {
+            static constexpr auto timeout_seconds = 3;
+            static constexpr auto input_buffer_size = 2048;
+            static constexpr auto max_retry_count = 10;
             const auto query = create_query<T>(domain);
             auto buf = asio::streambuf{};
             auto out = std::ostream{&buf};
             out << query;
-            co_await socket_.async_send(buffer(buf.data(), buf.size()), asio::use_awaitable);
 
-            auto input = std::array<char, 1024>{};
-            co_await socket_.async_receive(asio::buffer(input), asio::use_awaitable);
+            auto input = std::array<char, input_buffer_size>{};
+            auto timer = asio::steady_timer{co_await asio::this_coro::executor};
+            boost::system::error_code receive_ec;
+            auto current_retry_count = 0;
+            do {
+                co_await socket_.async_send(buffer(buf.data(), buf.size()), asio::use_awaitable);
+                ++current_retry_count;
+                timer.expires_after(std::chrono::seconds(timeout_seconds));
+                std::tie(std::ignore, receive_ec, std::ignore, std::ignore) =
+                        co_await asio::experimental::make_parallel_group(
```

Comment:
There is a nicer way to make a parallel group using overloaded operators

Reply:
Could you tell me how exactly? I couldn't find anything in the documentation. Also can you tell the overloaded operators to `wait_for_one` like I did here?

Reply:
There is an example in the [video](https://youtu.be/icgnqFM-aY4)